src/NHibernate.Test/NHSpecificTest/GH1881/TestFixture.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System;
+using System.Linq;
+using System.Reflection;
+using NHibernate.Collection;
+using NHibernate.Collection.Generic;
+using NHibernate.Proxy.DynamicProxy;
+using NUnit.Framework;
+
+namespace NHibernate.Test.NHSpecificTest.GH1881
+{
+	[TestFixture]
+	[Obsolete]
+	public class TestFixture
+	{
+		[Test]
+		public void InterfacesShouldBeImplementedExplicitlyOnProxies()
+		{
+			var proxy = new ProxyFactory().CreateProxy(typeof(PersistentGenericBag<object>), null, typeof(ILazyInitializedCollection));
```

Comment:
Well... no, but that's kind of the point.
To reproduce the issue I would need a type that isn't validatable.

```
public interface IFoo
{
    bool Prop { get; }
}

public class Bar : IFoo
{
    public bool Prop => false;
}

// throws the same error
new ProxyFactory().CreateProxy(typeof(Bar), null, typeof(IFoo));
```

The proxy factory generates a proxy type for `Bar`, skipping `Prop` because it isn't proxyable.
But it adds the property from `IFoo`, which is proxyable, which is wrong and causes errors.
With the explicit implementation of the interface the proxy is valid for the interface, if not for the base type.

Reply:
It would be still better to use this type for the test than this big thing that `PersistentGenericBag` is.

So, there are actually no way to reproduce the failure when using the dynamic proxy factory only through NHibernate's proxy handling, isn't it? (Or does Envers disable the default proxy validation in NHibernate? There is a setting allowing this.)

So this change looks needed for somewhat external users of the dynamic proxy factory, using it for their own needs with less strict prerequisites on proxied classes. But that is not the purpose of NHibernate to provide a general purpose proxifying mechanism. In this spirit the new static proxy factory API is way less exposed (more internal), and the old dynamic proxy should be gone in 6.0.

Maybe Envers will need to use an external proxy library rather than using the NHibernate internal one. (Unless it can be changed to use the static proxy factory, if it is not too specific to NHibernate for being used for Envers case. But I doubt about this.)

(In other words, this all looks to me at lot like this other [discussion](https://github.com/nhibernate/nhibernate-core/pull/1709#commitcomment-29603179).)

Reply:
In the example above `Bar` will fail the validation.

Reply:
> Or does Envers disable the default proxy validation in NHibernate? There is a setting allowing this.

They use it for something different - as a general purpose proxy factory to inject an interceptor, not for nhibernate entities.